src/commonMain/kotlin/app/cash/turbine/Turbine.kt

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (C) 2022 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.turbine
+
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.cancelAndJoin
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.channels.Channel.Factory.UNLIMITED
+import kotlinx.coroutines.channels.ChannelResult
+
+internal const val debug = false
+
+/**
+ * A standalone [Turbine] suitable for usage in fakes or other external test components.
+ */
+public interface Turbine<T>: ReceiveTurbine<T>  {
+  /**
+   * Returns the underlying [Channel].
+   */
+  public override fun asChannel(): Channel<T>
+
+  public suspend fun cancel(cause: Throwable?)
```

Comment:
Default of null? Is that how you send a close event? The receive API still has item/close/error functions so we need to ensure there's an easy way to understand how to send those events.

Reply:
I think that renaming this to `close` may be the right choice. I will need to review the code and see what I was thinking with this visibility approach in the first place before I can say for certain, though

Reply:
Okay, this is me being deep in the weeds on cancel vs. close:

* Implementation currently in the PR uses `close` *all* the time. No matter what. I did this because it worked, didn't think too deeply about it
* I named these all `cancel` because the original was named cancel
* The original was named cancel because it cancelled an underlying job

So:
* This is a mess

Plan:
* Keep `ReceiveTurbine.cancel()`. This will continue to be parameterless, because `cancelAndJoin` is parameterless. `cancel()` will also invoke `cancel()` on the underlying `Channel`, so this will result in a `Turbine` that produces `Error(CancellationException())`. Makes sense to me, because that's what you get when you call `receive()` on a cancelled channel
* Add `Turbine.close(cause: Throwable? = null)`. Separate code path from `ReceiveTurbine.cancel()`. This should also cancel the associated job, but will not `suspend`.